scripts/generate-modern-sbom.sh

```diff
@@ -0,0 +1,196 @@
+    if ! command -v syft >/dev/null && [ ! -f "${syft_bin}" ]; then
+        log_callout "installing syft to local bin directory..."
+        mkdir -p ./bin
+        curl -sSfL https://raw.githubusercontent.com/anchore/syft/main/install.sh | sh -s -- -b ./bin
+        syft_bin="./bin/syft"
+    elif [ -f "${syft_bin}" ]; then
+        log_callout "syft is already installed locally: $(${syft_bin} version --output json | jq -r '.version')"
+    else
+        log_callout "syft is already installed: $(syft version --output json | jq -r '.version')"
+        syft_bin="syft"
+    fi
+    
+    # export the syft binary path for use in other functions
+    export SYFT_BIN="${syft_bin}"
+}
+
+function install_sbomasm() {
+    local sbomasm_bin="./bin/sbomasm"
+    if ! command -v sbomasm >/dev/null && [ ! -f "${sbomasm_bin}" ]; then
+        log_callout "installing sbomasm to local bin directory..."
+        mkdir -p ./bin
+        
+        # determine architecture and OS for sbomasm download
+        local os arch
+        os=$(uname -s | tr '[:upper:]' '[:lower:]')
+        arch=$(uname -m)
+        
+        case "${arch}" in
+            x86_64) arch="amd64" ;;
+            aarch64|arm64) arch="arm64" ;;
+            *) log_error "unsupported architecture: ${arch}"; return 1 ;;
+        esac
+        
+        # download latest sbomasm release
+        local download_url="https://github.com/interlynk-io/sbomasm/releases/latest/download/sbomasm-${os}-${arch}"
+        if curl -sSfL "${download_url}" -o "${sbomasm_bin}"; then
+            chmod +x "${sbomasm_bin}"
+            log_success "sbomasm installed successfully"
+        else
+            log_error "Failed to download sbomasm"
+            return 1
+        fi
+    elif [ -f "${sbomasm_bin}" ]; then
+        log_callout "sbomasm is already installed locally"
+    else
+        log_callout "sbomasm is already installed: $(sbomasm version 2>/dev/null || echo 'system')"
+        sbomasm_bin="sbomasm"
+    fi
+    
+    # export the sbomasm binary path for use in other functions
+    export SBOMASM_BIN="${sbomasm_bin}"
+}
+
+function augment_sbom() {
+    local version="$1"
+    local output_dir="$2"
+    
+    log_callout "augmenting SBOM files with enhanced metadata..."
+    
+    local sbomasm_cmd="${SBOMASM_BIN:-sbomasm}"
+    
+    # augment SPDX SBOM with etcd-specific metadata
+    log_callout "augmenting SPDX SBOM..."
+    if "${sbomasm_cmd}" edit --append --subject Document \
+        --author 'etcd maintainers (https://github.com/etcd-io/etcd/blob/main/MAINTAINERS)' \
+        --supplier 'etcd project (https://etcd.io/)' \
+        --repository 'https://github.com/etcd-io/etcd' \
+        --license 'Apache-2.0 (https://raw.githubusercontent.com/etcd-io/etcd/main/LICENSE)' \
+        "${output_dir}/sbom.spdx.json" > "${output_dir}/sbom.spdx.json.tmp"; then
+        mv "${output_dir}/sbom.spdx.json.tmp" "${output_dir}/sbom.spdx.json"
+        log_success "SPDX SBOM augmented successfully"
```

Comment:
When running `make verify` locally on this pr I see:

```bash
augmenting SBOM files with enhanced metadata...                                                                                                               
augmenting SPDX SBOM...                                    
{"level":"info","ts":1752054550.5096443,"caller":"edit/spdx_edit.go:82","msg":"SPDX error updating supplier: not supported"}
{"level":"info","ts":1752054550.5097733,"caller":"edit/spdx_edit.go:82","msg":"SPDX error updating repository: not supported"}
```

Are the errors on updating repository and supplier expected?

Reply:
In this case, I would hold the pull request until they release the new version (assuming we have verified that this addresses the issue).